### ISSUE TITLE
Initial load locks up browser in default onOptionAdd

### ISSUE THREAD

**Issue by StuAtGit** [machineboy2045/angular-selectize#105]
Once loaded, selectize can handle a large number of options (>10K), but it does lock up the browser on initial load, and it looks like the lock up is caused by the default onOptionAdd method:

```
settings.onOptionAdd = function(value, data) {
        if( scope.options.indexOf(data) === -1 ) {
          scope.options.push(data);

          if (scope.config.onOptionAdd) {
            scope.config.onOptionAdd.apply(this, arguments);
          }
        }
      };
```

angular ui-grid handles the exact same data without locking up the browser, so there must be some way of loading it into a component, without locking everything up.

I understand this is pushing things fairly hard - I expect the load to take a bit. However, it would be nice if the load didn't lock up the whole browser (Should I overload the onOptionAdd with function  that invoked my own web worker or something?). I understand this is a special case - I'm more looking for recommendations than a "fix" of some kind to selectize. For example, some documentation on how to override the default behaviour of onOptionAdd to handle large number of options without locking up the browser.

Worst-case I could just use a ui-grid with a search header, and one row as a sort of not-as-nice-looking selectize, but if there's a way to do it with this component, that would be great.

**Comment by StuAtGit** [comment]
I looked into it a little more, and just overloading onOptionAdd won't work - since what I was calling the "default" onOptionAdd is a method that is always called, and then invokes any user-configured onOptionAdd on the arguments.

It also looks like it's not simply the size of the data - selectize is performing various (what look like dom manipuations) and check in onOptionAdd and setSelectizeOptions that eat up CPU when you're iterating over a large list. Will post as I learn more.

**Comment by machineboy2045** [comment]
If you add all 10k options at once, then Selectize builds all the dom
elements at once. You could possibly get around this by pushing your
options 1k at a time into $scope.options, with a timeout in-between each
batch.

**Comment by StuAtGit** [comment]
Actually tried that yesterday, the timeout doesn't seem to help. Locks up just as bad. I added them in batches of 100, and even dumped logs showing they were added in batches to verify that's what I was doing. I also tried ripping out some of the existence checks in onOptionAdd code, but the perf problem just moved to setSelectizeOptions. 

I think the problem basically comes down to having tens of thousands of option elements attached to the Dom tree. Which I think means what I really want is just a search/typeahead widget that isn't adding anything all the items into the Dom. Selectize was an easy, neat way of effectively getting what I can best describe as a typeahead search with a results popup... but easy may not be correct here ;)
